selectors/selectors.md

```diff
@@ -110,295 +110,269 @@ These components imply or expand into the following things:
-    // Use this for all human readable input/output (terminal, etc).
-    String() string
+It will be a Recursive Selector following a long a certain field (`prev` in this case):
+
+```json
+{
+  "cidRootedSelector": {
+    "root": "cidabcdef",
+    "selectors": [
+      {"selectCid": "cidabcdef"},
+      {"selectRecursive": {
+        "follow": [
+          {"selectPath": "prev"}
+        ],
+        "cidLimit": "somecid"
+      }}
+    ]
+  }
 }
+```
+
+
+### Getting a full sub-DAG
 
-// SelectorType is the enum used for selector type codes. These type codes
-// are part of the protocol and housed in the multicodec table.
-type SelectorType int
-const (
-    NullSelector SelectorType = 0
-    CidSelector SelectorType = 1
-    PathSelector SelectorType = 2
-)
-
-// Select is a function that applies a selector to a dag, returning the resulting
-// subdag. It is implemented using a Traverser. Given a lazy traverser
-// implementation, it should be a lazy application and avoid visiting the whole
-// dag until it needs to.
-func Select(d Dag, s Selector) Dag {
-    t := d.Traverse()
-    t.Filter(s)
-    return &traversedDag{d, s, t}
+For getting a full file from UnixFSv1 you need to retrieve a full sub-DAG.
+
+An example selector to get the full sub-DAG rooted at a certain CID:
+
+
+```json
+{
+  "cidRootedSelector": {
+    "root": "cidabcdef",
+    "selectors": [
+      {"selectRecursive": {
+        "follow": [
+          {"selectPath": "Links"},
+          {"selectArrayAll": null},
+          {"selectPath": "multihash"}
+        ]
+      }}
+    ]
+  }
 }
+```
 
-// traversedDag is a utility dag implementation that uses a traverser to
-// to apply a selector. this is what Select uses.
-type traversedDag struct {
-    d Dag
-    s Selector
-    t Traverser
+If it's a file in some directory, you can also start at a deeper level:
+
+```json
+{
+  "cidRootedSelector": {
+    "root": "cidabcdef",
+    "selectors": [
+      {"selectCid": "cidabcdef"},
+      {"selectPath": "with"},
+      {"selectPath": "some"},
+      {"selectPath": "subdirectory"},
+      {"selectRecursive": {
+        "follow": [
+          {"selectPath": "Links"},
+          {"selectArrayAll": null},
+          {"selectPath": "multihash"}
+        ]
+      }}
+    ]
+  }
 }
+```
 
-func (d *traversedDag) Root() Node {
-    if d.s.IncludesPath(RootPath) {
-        return d.d.Root()
-    }
-    return nil
+IPLD Schema
+-----------
+
+```ipldsch
+# This is the main entry point for the current selectors
+type CidRootedSelector struct {
+  root Cid
+  # Each element matches a path segement (or several, in some cases)
+  selectors [Selector]
 }
 
-func (d *traversedDag) GetNode(path Path) Node {
-    if d.s.IncludesPath(path) {
-        return d.d.GetNode(path)
-    }
-    return nil
+# A catch all for all types of selectors. They are split between recursive and
+# non-recursive ones as a recursive selector can't have another recursive
+# selector embedded
+type Selector union {
```

Comment:
you define selector types here, but not the functions the selector exposes (the string/byte representation, the type, etc) - intentional?

Reply:
The functionality is described above. This is a description on how they are encoded within the IPLD data model.